Cython/Utility/ImportExport.c

```diff
@@ -1,211 +1,95 @@
-/////////////// ImportLookup.proto ///////////////
+/////////////// Import.proto ///////////////
 
-static PyObject *__Pyx__Import_Lookup(PyObject *name); /*proto*/
-static PyObject *__Pyx__Import_LookupLock(PyObject *name); /*proto*/
+static PyObject *__Pyx_Import(PyObject *name, PyObject *from_list, PyObject *qualname, int level); /*proto*/
 
-/////////////// ImportLookup ///////////////
+/////////////// Import ///////////////
+//@requires: StringTools.c::IncludeStringH
+//@requires: Builtins.c::HasAttr
 
-static PyObject *__Pyx__Import_Lookup(PyObject *name) {
+static PyObject *__Pyx__Import_GetModuleFromSysModules(PyObject *qualname) {
     PyObject *imported_module;
 #if (CYTHON_COMPILING_IN_PYPY && PYPY_VERSION_NUM  < 0x07030400) || \
         CYTHON_COMPILING_IN_GRAAL
     PyObject *modules = PyImport_GetModuleDict();
     if (unlikely(!modules))
         return NULL;
-    imported_module = __Pyx_PyDict_GetItemStr(modules, name);
+    imported_module = __Pyx_PyDict_GetItemStr(modules, qualname);
     Py_XINCREF(imported_module);
 #else
-    imported_module = PyImport_GetModule(name);
+    imported_module = PyImport_GetModule(qualname);
```

Comment:
I think the `PyImport_GetModule` is essentially forcing the "has this been fully imported" check that the removed `"__spec__"` check used to enforce.

That suggests to me that the PyPy/Graal code above needs to do the same. With that said, I don't think we support PyPy earlier than that, and Graal looks to have implemented it too (and I'm not sure it's worth defining much of a support window on that yet) so I think remove the code above rather than try to make it work.